### PR TITLE
expose consturct method; customizable console.*;

### PR DESCRIPTION
**Pull request by ialex32x**

### CODE REVIEW

**Comment by FabianTe** [comment]
+1 On the abillity to supply a custom console. Considering using otto but this is still one blocker for me.

**Comment by andig** [comment]
+1 for the console changes. Not sure about the construct method, that looks like a separate PR.

@ialex32x would you split this PR? @deoxxa would you consider this for merging?

**Comment by frankffenn** [comment]
+1 I love this

File: otto.go [hunk 1]
```diff
@@ -598,6 +598,16 @@ func (self Otto) Call(source string, this interface{}, argumentList ...interface
 	}
 }
 
+// ialex32x: provide a method to construct js object by golang
```
Comment (stevenh) [inline otto.go:601]:
```suggestion
// Construct provides a method to construct js object by golang.
```

**Review by stevenh — CHANGES_REQUESTED** [review]
Could you rebase this and run a go mod tidy to remove the reference to readline.

**Comment by stevenh** [comment]
Closing due to age with no feedback

**Comment by andig** [comment]
@stevenh there is a lot of positiv feedback. Would you want me to merge this approach into #445?

**Pull request closed without merging**